dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala

```diff
@@ -429,11 +428,21 @@ abstract class DLCWallet
             witnessScriptOpt = InputInfo.getScriptWitness(utxo.inputInfo)
           )
       }
-
       _ = logger.info(
         s"Created offer with tempContractId ${offer.tempContractId.hex}")
-      _ <- dlcInputsDAO.createAll(dlcInputs)
-      _ <- dlcOfferDAO.create(dlcOfferDb)
+      globalAction = dlcDAO.createAction(dlcDb)
+      contractAction = contractDataDAO.createAction(contractDataDb)
+      announcementAction = dlcAnnouncementDAO.createAllAction(
+        dlcAnnouncementDbs)
+      inputsAction = dlcInputsDAO.createAllAction(dlcInputs)
+      offerAction = dlcOfferDAO.createAction(dlcOfferDb)
+      actions = Vector(globalAction,
+                       contractAction,
+                       announcementAction,
+                       inputsAction,
+                       offerAction)
+      allActions = DBIO.sequence(actions)
+      _ <- safeDatabase.run(allActions)
```

Comment:
Maybe should make a `DLCDbActionManager` (or a better name) trait that contains all the long string of actions

Reply:
I'm not sure if I understand, I can see a helper method to reduce clutter but i think this action is contextual to `createDLCOffer()`? Is this used else where too?

Reply:
Yeah mostly to reduce clutter